src/java/org/apache/cassandra/tools/nodetool/layout/CassandraCliHelpLayout.java

```diff
@@ -0,0 +1,786 @@
+                for (int i = 0; i < optionSpec.description().length; i++)
+                {
+                    if (i == 0)
+                    {
+                        table.addRowValues(leadingSpaces.concat(
+                            colorScheme().optionText(String.format(SUBCOMMAND_OPTION_TEMPLATE,
+                                                                   optionSpec.longestName(),
+                                                                   colorScheme().text(optionSpec.description()[i])))));
+                    }
+                    else
+                    {
+                        table.addRowValues(leadingSpaces.concat(colorScheme().text(optionSpec.description()[i])));
+                    }
+                }
+            }
+        }
+        return table.toString();
+    }
+
+    @Override
+    public String footerHeading(Object... params)
+    {
+        return createHeading(FOOTER_HEADING, params);
+    }
+
+    @Override
+    public String footer(Object... params)
+    {
+        String[] footer;
+        if (commandSpec().parent() == null)
+            footer = isEmpty(commandSpec().usageMessage().footer()) ? new String[]{ USAGE_HELP_FOOTER } :
+                     commandSpec().usageMessage().footer();
+        else
+            footer = EMPTY_FOOTER;
+
+        TextTable table = TextTable.forColumns(colorScheme(), new Column(commandSpec().usageMessage().width(), 0, Column.Overflow.WRAP));
+        table.setAdjustLineBreaksForWideCJKCharacters(commandSpec().usageMessage().adjustLineBreaksForWideCJKCharacters());
+        table.indentWrappedLines = 0;
+
+        for (String summaryLine : footer)
+            table.addRowValues(String.format(summaryLine, params));
+        table.addEmptyRow();
+        return table.toString();
+    }
+
+    public String topLevelCommandListHeading(Object... params)
+    {
+        return createHeading(TOP_LEVEL_COMMAND_HEADING + "%n", params);
+    }
+
+    public String topLevelSynopsis(Object... params)
+    {
+        return printDetailedSynopsis(commandSpec(), TOP_LEVEL_SYNOPSIS_LIST_PREFIX, COLUMN_INDENT, false);
+    }
+
+    /**
+     * Returns the help for the top-level command. This differs from the {@link #commandList(Map)} method
+     * in that it does not include the subcommands.
+     * <pre>
+     * {@code
+     * The most commonly used nodetool commands are:
+     *     abortbootstrap                      Abort a failed bootstrap
+     *     bootstrap                           Monitor/manage node's bootstrap process
+     *     cidrfilteringstats                  Print statistics on CIDR filtering
+     *     clientstats                         Print information about connected clients
+     * }
+     * </pre>
+     * @param params Arguments referenced by the format specifiers in the header strings.
+     * @return The top-level subcommands list.
+     */
+    public String topLevelCommandList(Object... params)
```

Comment:
`params` is unused?

Reply:
this is actually an implementation of the `IHelpSectionRenderer`, so the signature should be `String render(Help help)`.

```
    public interface IHelpSectionRenderer {
        /**
         * Renders a section of the usage help, like header heading, header, synopsis heading,
         * synopsis, description heading, description, etc.
         * @param help the {@code Help} instance for which to render a section
         * @return the text for this section; may contain {@linkplain Help.Ansi ANSI} escape codes
         * @since 3.9
         */
        String render(Help help);
    }
```